library/src/amd_detail/hipsolver.cpp

```diff
@@ -607,6 +607,41 @@ catch(...)
     return hipsolver::exception2hip_status();
 }
 
+hipsolverStatus_t hipsolverSetDeterministicMode(hipsolverHandle_t            handle,
+                                                hipsolverDeterministicMode_t mode)
+try
+{
+    if(!handle)
+        return HIPSOLVER_STATUS_NOT_INITIALIZED;
+
+    return hipsolver::rocblas2hip_status(rocblas_set_atomics_mode(
```

Comment:
Will this affect only rocblas or will it also need something like "rocsolver_set_atomics_mode()"?    Just checking.

Reply:
To my knowledge, rocSOLVER doesn't use any non-deterministic programming patterns, so any non-determinism is either a bug or coming from rocBLAS. If we start using some non-deterministic patterns then we'll need to add something similar to rocSOLVER to be called by this method.

Reply:
It seems to me atomic operations are used in some source codes. 
Perhaps these are safe uses  on integer types  and so won't introduce non-deterministic behaviour?
Just checking.

src/lapack/roclapack_syevj_heevj.hpp:            atomicAdd(completed, 1);
src/lapack/roclapack_syevj_heevj.hpp:            atomicAdd(completed, 1);
src/refact/rocrefact_csrrf_splitlu.hpp:        atomicAdd(&(nzUarray[irow]), lnnzUrow);
src/refact/rocrefact_csrrf_splitlu.hpp:        atomicAdd(&(nzLarray[irow]), (lid == 0) ? 1 + lnnzLrow : lnnzLrow);
src/refact/rocrefact_csrrf_splitlu.hpp:        atomicAdd(&(nnzUrow[irow]), lnnzUrow);
src/include/lib_device_helpers.hpp:                _info_temp = atomicCAS(&_info, _info_temp, i + 1);
src/specialized/roclapack_getri_specialized_kernels.hpp:                _info_temp = atomicCAS(&_info, _info_temp, i + 1);